notebooks/03-query/07-functions-datetime.ipynb

```diff
@@ -0,0 +1,963 @@
+    "SELECT\n",
+    "  TIME_PARSE(\"timestamp\") AS \"__time\",\n",
+    "  \"browser\",\n",
+    "  \"browser_version\",\n",
+    "  \"city\",\n",
+    "  \"continent\",\n",
+    "  \"country\",\n",
+    "  \"version\",\n",
+    "  \"event_type\",\n",
+    "  \"event_subtype\",\n",
+    "  \"loaded_image\",\n",
+    "  \"session\",\n",
+    "  \"session_length\",\n",
+    "  TIME_SHIFT(TIME_PARSE(\"timestamp\"), 'PT0.01S', \"session_length\") AS \"ended_at\"\n",
+    "FROM \"ext\"\n",
+    "PARTITIONED BY DAY\n",
+    "'''\n",
+    "\n",
+    "display.run_task(sql)\n",
+    "sql_client.wait_until_ready('example-koalas-fndatetime')\n",
+    "display.table('example-koalas-fndatetime')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ba4eeec6-5e7e-4524-a203-4a25cc975d90",
+   "metadata": {},
+   "source": [
+    "Time functions are available in both SQL and [native JSON ingestion](https://druid.apache.org/docs/27.0.0/querying/math-expr#time-functions) specifications. \n",
+    "\n",
+    "Address this new timestamp in the table, [stored as a `LONG`](https://druid.apache.org/docs/27.0.0/querying/sql-data-types#standard-types), using `MILLIS_TO_TIMESTAMP`.\n",
+    "\n",
+    "Run the next cell, showing `MILLIS_TO_TIMESTAMP` being used alongside `TIME_IN_INTERVAL`. Notice these results are the same as are produced by the SQL statement above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9630abff-f406-4cf6-b846-f03145d40a30",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "SELECT\n",
+    "  __time AS \"start\",\n",
+    "  MILLIS_TO_TIMESTAMP(\"ended_at\") AS \"end\"\n",
+    "FROM \"example-koalas-fndatetime\"\n",
+    "WHERE TIME_IN_INTERVAL(MILLIS_TO_TIMESTAMP(\"ended_at\"),'2019-08-25T13/PT1M')\n",
+    "AND TIME_IN_INTERVAL(__time, 'PT1M/2019-08-25T13')\n",
+    "LIMIT 10\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "510f208b-fdaa-45fe-9908-c6ebbe4c754c",
+   "metadata": {},
+   "source": [
+    "## Use functions to format the date and time\n",
+    "\n",
+    "Below are examples of:\n",
+    "\n",
+    "* Using `TIME_FORMAT` to apply a string pattern to a timestamp.\n",
+    "* Extracting elements of a timestamp using `TIME_EXTRACT` and `EXTRACT`.\n",
+    "\n",
+    "The `TIME_FORMAT` function applies a [JODA](https://www.joda.org/joda-time/apidocs/org/joda/time/format/DateTimeFormat.html) picture to a datetime column.\n",
```

Comment:
picture or pattern/template?

Reply:
You're right – their docs even say "pattern".